src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,98 @@
 package core.basesyntax;
 
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+
+    private Node<K, V>[] table;
+    private int size;
+    private int capacity;
+    private final float loadFactor;
+    private int threshold;
+
+    private static class Node<K, V> {
+        final K key;
+        V value;
+        Node<K, V> next;
+
+        public Node(K key, V value, Node<K, V> next) {
+            this.key = key;
+            this.value = value;
+            this.next = next;
+        }
+    }
+
+    public MyHashMap() {
+        this.capacity = DEFAULT_CAPACITY;
+        this.loadFactor = DEFAULT_LOAD_FACTOR;
+        this.table = new Node[capacity];
+        this.threshold = (int) (capacity * loadFactor);
+    }
 
     @Override
-    public void put(K key, V value) {
+    public int getSize() {
+        return size;
+    }
+
+    private int hash(K key) {
+        return key == null ? 0 : Math.abs(key.hashCode() % capacity);
+    }
+
+    private void resize() {
+        capacity = capacity * 2;
+        Node<K, V>[] newTable = new Node[capacity];
+        threshold = (int) (capacity * loadFactor);
 
+        for (int i = 0; i < table.length; i++) {
+            Node<K, V> node = table[i];
+            while (node != null) {
+                Node<K, V> next = node.next;
+                int newIndex = hash(node.key);
+                node.next = newTable[newIndex];
+                newTable[newIndex] = node;
+                node = next;
+            }
+        }
+        table = newTable;
     }
 
     @Override
-    public V getValue(K key) {
-        return null;
+    public void put(K key, V value) {
+        int index = hash(key);
+        Node<K, V> node = table[index];
+
+        if (node == null) {
+            table[index] = new Node<>(key, value, null);
+            size++;
+        } else {
+            Node<K, V> prev = null;
+            while (node != null) {
+                if (node.key.equals(key)) {
```

Comment:
Remember to check for null before calling equals on key to avoid potential NullPointerException. Use Objects.equals(node.key, key) for null safety.